### PR TITLE
re-organzied port-agent and added argocd installation method

### PR DESCRIPTION
**Pull request by stavbernazport**
# Description

re-organzied port-agent and added argocd installation method

## Added docs pages

Please also include the path for the added docs

- helm method (`/create-self-service-experiences/setup-backend/webhook/port-execution-agent/installation-methods/helm.md`)
- argocd method (`/create-self-service-experiences/setup-backend/webhook/port-execution-agent/installation-methods/argocd.md`)
- methods (`/create-self-service-experiences/setup-backend/webhook/port-execution-agent/installation-methods/installation-methods.md`)
- usage (`/create-self-service-experiences/setup-backend/webhook/port-execution-agent/usage.md`)
- main category (`/create-self-service-experiences/setup-backend/webhook/port-execution-agent/port-execution-agent.md`)

### CODE REVIEW

File: docs/create-self-service-experiences/setup-backend/webhook/port-execution-agent/installation-methods/argocd.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      - CreateNamespace=true
+```
+
+</details>
+<br/>
+
+
+## Next Steps
+
+- [Usage](/create-self-service-experiences/self-service-actions-deep-dive/self-service-actions-deep-dive.md) guide Set up a webhook.
```
Comment (hadar-co) [inline docs/create-self-service-experiences/setup-backend/webhook/port-execution-agent/installation-methods/argocd.md:74]: This is unclear - what is the step? The link just takes me to the deep dive page. Also, the sentence is incorrect - "Usage guide set up a webhook". Please revise:)

File: docs/create-self-service-experiences/setup-backend/webhook/port-execution-agent/installation-methods/argocd.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
+---
+sidebar_position: 2
+---
+
+# ArgoCD
+
+Let's dive in to a walkthrough on how to install the Port execution agent in your Kubernetes cluster.
```
Comment (hadar-co) [inline docs/create-self-service-experiences/setup-backend/webhook/port-execution-agent/installation-methods/argocd.md:7]: You used the same sentence here and in the Helm page. Perhaps let's change it to something like:
This page will walk you through the installation of the Port execution agent in your Kubernetes cluster using ArgoCD.

And let's use the same format in the Helm page.

File: docs/create-self-service-experiences/setup-backend/webhook/port-execution-agent/installation-methods/helm.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
+---
+sidebar_position: 1
+---
+
+# Helm
+
+Let's dive in to a walkthrough on how to install the Port execution agent in your Kubernetes cluster.
```
Comment (hadar-co) [inline docs/create-self-service-experiences/setup-backend/webhook/port-execution-agent/installation-methods/helm.md:7]: See comment in argocd.md line 7

File: docs/create-self-service-experiences/setup-backend/webhook/port-execution-agent/installation-methods/argocd.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+:::info
+You can observe the helm chart with the full installation [here](https://github.com/port-labs/helm-charts/tree/main/charts/port-agent).
+:::
+
+:::note prerequisites
+
+- [ArgoCD](https://argoproj.github.io/cd/) must be installed in your Kubernetes cluster. Please refer to
+  ArgoCD's [documentation](https://argo-cd.readthedocs.io/en/stable/) for further details on the installation;
```
Comment (hadar-co) [inline docs/create-self-service-experiences/setup-backend/webhook/port-execution-agent/installation-methods/argocd.md:16]:
```suggestion
- [ArgoCD](https://argoproj.github.io/cd/) must be installed in your Kubernetes cluster. Please refer to the [ArgoCD documentation](https://argo-cd.readthedocs.io/en/stable/) for further details about the installation.
```

File: docs/create-self-service-experiences/setup-backend/webhook/port-execution-agent/installation-methods/helm.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+:::info
+You can observe the helm chart with the full installation [here](https://github.com/port-labs/helm-charts/tree/main/charts/port-agent).
+:::
+
+:::note prerequisites
+
+- The [Helm](https://helm.sh) must be installed to use the chart. Please refer to
```
Comment (hadar-co) [inline docs/create-self-service-experiences/setup-backend/webhook/port-execution-agent/installation-methods/helm.md:15]:
```suggestion
- [Helm](https://helm.sh) must be installed to use the chart. Please refer to
```

File: docs/create-self-service-experiences/setup-backend/webhook/port-execution-agent/installation-methods/helm.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  the Helm's [documentation](https://helm.sh/docs) for further details on the installation;
```
Comment (hadar-co) [inline docs/create-self-service-experiences/setup-backend/webhook/port-execution-agent/installation-methods/helm.md:16]:
```suggestion
  the [Helm documentation](https://helm.sh/docs) for further details about the installation.
```

File: docs/create-self-service-experiences/setup-backend/webhook/port-execution-agent/installation-methods/helm.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+```bash showLineNumbers
+helm repo add port-labs https://port-labs.github.io/helm-charts
+```
+
+If you already added this repo earlier, run `helm repo update` to retrieve
+the latest versions of the charts. You can then run `helm search repo port-labs` to see the charts.
+
+2. Install the `port-agent` chart by using the following command:
+
+## Installation
```
Comment (hadar-co) [inline docs/create-self-service-experiences/setup-backend/webhook/port-execution-agent/installation-methods/helm.md:33]: Move the "Installation" title up, it should be directly below the prerequisites like you did in argocd.md

File: docs/create-self-service-experiences/setup-backend/webhook/port-execution-agent/installation-methods/helm.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+    --create-namespace --namespace port-agent \
+    --set env.normal.PORT_ORG_ID=YOUR_ORG_ID \
+    --set env.normal.KAFKA_CONSUMER_GROUP_ID=YOUR_KAFKA_CONSUMER_GROUP \
+    --set env.secret.PORT_CLIENT_ID=YOUR_PORT_CLIENT_ID \
+    --set env.secret.PORT_CLIENT_SECRET=YOUR_PORT_CLIENT_SECRET
+```
+
+## Next Steps
+
+- [Usage](/create-self-service-experiences/self-service-actions-deep-dive/self-service-actions-deep-dive.md) guide Set up a webhook.
```
Comment (hadar-co) [inline docs/create-self-service-experiences/setup-backend/webhook/port-execution-agent/installation-methods/helm.md:50]: See comment at the end of argocd.md

File: docs/create-self-service-experiences/setup-backend/webhook/port-execution-agent/installation-methods/installation-methods.md [hunk 9]
```diff
@@ -0,0 +1,3 @@
+# Installation Methods
```
Comment (hadar-co) [inline docs/create-self-service-experiences/setup-backend/webhook/port-execution-agent/installation-methods/installation-methods.md:1]: You can delete this page entirely, it isnt needed, the "Installation methods" can just be a category in the sidebar but without a page. If you're not sure how to do this talk to me:)

File: docs/create-self-service-experiences/setup-backend/webhook/port-execution-agent/port-execution-agent.md [hunk 10]
```diff
@@ -26,5 +26,5 @@ The data flow when using the Port execution agent is as follows:
 
 ## Next steps
 
-- [Explore How to install and use the agent](./Installation.md)
-- [Control the payload sent to your endpoint](./port-execution-agent.md)
+- [Explore How to install and use the agent](/create-self-service-experiences/setup-backend/webhook/port-execution-agent/installation-methods/installation-methods.md)
```
Comment (hadar-co) [inline docs/create-self-service-experiences/setup-backend/webhook/port-execution-agent/port-execution-agent.md:29]: If we remove the "Installation methods" page like i suggested in my previous comment, this link will not work. So we can replace it with 2 links - one for Helm and one for ArgoCD.

**Review by hadar-co — CHANGES_REQUESTED** [review]

File: docs/create-self-service-experiences/setup-backend/webhook/port-execution-agent/installation-methods/helm.md [hunk 11]
```diff
@@ -0,0 +1,50 @@
+---
+sidebar_position: 1
+---
+
+# Helm
+
+Let's dive in to a walkthrough on how to install the Port execution agent in your Kubernetes cluster.
+
+:::info
+You can observe the helm chart with the full installation [here](https://github.com/port-labs/helm-charts/tree/main/charts/port-agent).
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-backend/webhook/port-execution-agent/installation-methods/helm.md:10]:
```suggestion
You can observe the helm chart and the available parameters [here](https://github.com/port-labs/helm-charts/tree/main/charts/port-agent).
```

File: docs/create-self-service-experiences/setup-backend/webhook/port-execution-agent/installation-methods/helm.md [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+  the Helm's [documentation](https://helm.sh/docs) for further details on the installation;
+- The connection credentials to Kafka are provided to you by Port.
+- If you want to trigger a GitLab Pipeline, you need to have a [GitLab trigger token](https://docs.gitlab.com/ee/ci/triggers/)
+
+:::
+
+1. Add Port's Helm repo by using the following command:
+
+```bash showLineNumbers
+helm repo add port-labs https://port-labs.github.io/helm-charts
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-backend/webhook/port-execution-agent/installation-methods/helm.md:25]: Let's add both the `add` and the `repo update` commands in the same code block. I don't want to rely on the user to remember that he installed in the past

File: docs/create-self-service-experiences/setup-backend/webhook/port-execution-agent/installation-methods/helm.md [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+```
+
+If you already added this repo earlier, run `helm repo update` to retrieve
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-backend/webhook/port-execution-agent/installation-methods/helm.md:28]: We can remove this part completely after adding the `helm repo update` command to the above code block

File: docs/create-self-service-experiences/setup-backend/webhook/port-execution-agent/installation-methods/argocd.md [hunk 14]
```diff
@@ -0,0 +1,74 @@
+---
+sidebar_position: 2
+---
+
+# ArgoCD
+
+Let's dive in to a walkthrough on how to install the Port execution agent in your Kubernetes cluster.
+
+:::info
+You can observe the helm chart with the full installation [here](https://github.com/port-labs/helm-charts/tree/main/charts/port-agent).
```
Comment (MPTG94) [inline docs/create-self-service-experiences/setup-backend/webhook/port-execution-agent/installation-methods/argocd.md:10]: I altered the sentence slightly in the comment on the helm page, let's apply it here too

**Review by MPTG94 — CHANGES_REQUESTED** [review]
Please go over both my comments and Hadar's

File: docs/create-self-service-experiences/setup-backend/webhook/port-execution-agent/installation-methods/argocd.md [hunk 15]
```diff
@@ -0,0 +1,90 @@
...
+    - CreateNamespace=true
+```
+
+</details>
+<br/>
+
+
+## Next Steps
+
+- Refer to the [Usage](/create-self-service-experiences/setup-backend/webhook/port-execution-agent/usage.md) guide Set up a webhook.
```
Comment (hadar-co) [inline docs/create-self-service-experiences/setup-backend/webhook/port-execution-agent/installation-methods/argocd.md:90]:
```suggestion
- Refer to the [usage guide](/create-self-service-experiences/setup-backend/webhook/port-execution-agent/usage.md) to set up a webhook.
```

**Review by hadar-co — APPROVED** [review]

**Pull request merged**